File: sdk/python/cachegpt/exceptions.py
"""
CacheGPT SDK Exceptions
"""


class CacheGPTError(Exception):
    """Base exception for all CacheGPT errors"""
    pass


class AuthenticationError(CacheGPTError):
    """Raised when API key is invalid or missing"""
    pass


class RateLimitError(CacheGPTError):
    """Raised when rate limit is exceeded"""

    def __init__(self, message: str, retry_after: int = None):
        super().__init__(message)
        self.retry_after = retry_after


class APIError(CacheGPTError):
    """Raised when API returns an error"""
    pass


class NetworkError(CacheGPTError):
    """Raised when network request fails"""
    pass


class ValidationError(CacheGPTError):
    """Raised when input validation fails"""
    pass

File: sdk/javascript/src/exceptions.ts
/**
 * Custom exception classes for CacheGPT SDK
 */

export class CacheGPTError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'CacheGPTError';
  }
}

export class APIError extends CacheGPTError {
  public status: number;
  public response?: any;

  constructor(message: string, status: number, response?: any) {
    super(message);
    this.name = 'APIError';
    this.status = status;
    this.response = response;
  }
}

export class NetworkError extends CacheGPTError {
  public cause?: Error;

  constructor(message: string, cause?: Error) {
    super(message);
    this.name = 'NetworkError';
    this.cause = cause;
  }
}

export class ValidationError extends CacheGPTError {
  constructor(message: string) {
    super(message);
    this.name = 'ValidationError';
  }
}

export class AuthenticationError extends APIError {
  constructor(message: string = 'Authentication failed') {
    super(message, 401);
    this.name = 'AuthenticationError';
  }
}

export class RateLimitError extends APIError {
  public retryAfter?: number;

  constructor(message: string, retryAfter?: number) {
    super(message, 429);
    this.name = 'RateLimitError';
    this.retryAfter = retryAfter;
  }
}